finality-aleph/src/sync/service.rs

```diff
@@ -177,24 +175,20 @@ where
         }
     }
 
-    fn perform_sync_action(&mut self, action: SyncAction<B, J>, peer: N::PeerId) {
-        use SyncAction::*;
-        match action {
-            Response(data) => self.send_to(data, peer),
-            HighestJustified(block_id) => self.request_highest_justified(block_id),
-            Noop => (),
-        }
-    }
-
     fn handle_state(&mut self, state: State<J>, peer: N::PeerId) {
+        use HandleStateAction::*;
         trace!(
             target: LOG_TARGET,
             "Handling state {:?} received from {:?}.",
             state,
             peer
         );
         match self.handler.handle_state(state, peer.clone()) {
-            Ok(action) => self.perform_sync_action(action, peer),
+            Ok(action) => match action {
```

Comment:
Could we go back to having this as a separate method?

Reply:
(after a short discussion) let's leave it that way